rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -273,14 +273,9 @@ if(WIN32)
 endif()
 if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
   install(TARGETS ${_target_name_lib}
-    EXPORT export_${_target_name_lib}
     ARCHIVE DESTINATION lib
     LIBRARY DESTINATION lib
     RUNTIME DESTINATION bin)
-
-  # Export this target so downstream interface packages can depend on it
-  rosidl_export_typesupport_targets("${rosidl_generator_py_suffix}" "${_target_name_lib}")
```

Comment:
Downstream interface packages depend on them. I'm not 100% sure they have to depend on them though.

https://github.com/ros2/rosidl_python/blob/3ce24baa074de225a314cbe8a286734e63482c96/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake#L264

That's using the special rosidl suffix'd targets variable. How about removing only the `ament_export_targets()`? That way downstream targets won't accidentally link against them if using `pkg_name_TARGETS` or `ament_target_dependencies()`.

Reply:
> Downstream interface packages depend on them. I'm not 100% sure they have to depend on them though.

Ah, I see.
I'm pretty sure that's not needed, but let me double check.

Reply:
> I checked the generated code, and yes a downstream interface package depends on the generated library.

Based on this, it seems that a bunch of stuff that I moved to `PRIVATE` in https://github.com/ros2/rosidl_python/pull/160 is not correct.
Why I did that is because `ament_export_dependencies()` doesn't allow you to specify components, so I worked that arround by not having to export anything at all.
But in retrospective that seems incorrect.

I have now exported the dependency of FindPython3 manually, creating a config file.

> That's using the special rosidl suffix'd targets variable. How about removing only the ament_export_targets()? That way downstream targets won't accidentally link against them if using pkg_name_TARGETS or ament_target_dependencies().

Unfortunately `rosidl_export_typesupport_targets()` requires you to also `ament_export_targets()`.
It would be nice if `ament_export_targets()` had an option to not append the target to pkg_name_TARGETS.
But considering that I have undone most of what I did in #160, I think we don't need to worry about that now.